src/compiler.rs

```diff
@@ -331,6 +333,51 @@ impl Compiler {
         block.add(Op::Constant(value), self.line());
     }
 
+    fn grouping_or_tuple(&mut self, block: &mut Block) {
+        let block_length = block.ops.len();
+        let token_length = self.curr;
+        if self.try_tuple(block).is_err() {
+            block.ops.truncate(block_length);
+            self.curr = token_length;
+            self.grouping(block);
+        }
+    }
+
+    fn try_tuple(&mut self, block: &mut Block) -> Result<(), ()> {
+        expect!(self, Token::LeftParen, "Expected '(' at start of tuple");
+
+        let mut num_args = 0;
+        loop {
+            match self.peek() {
+                Token::RightParen | Token::EOF => {
+                    break;
+                }
+                Token::Newline => {
+                    self.eat();
+                }
+                _ => {
+                    self.expression(block);
+                    num_args += 1;
+                    if self.peek() == Token::Comma {
+                        self.eat();
+                        continue;
+                    }
+                    if self.peek() == Token::RightParen {
+                        continue;
+                    }
+                    return Err(());
+                }
```

Comment:
```suggestion
                token => {
                    self.expression(block);
                    num_args += 1;
                    match token {
                        Token::Comma => { self.eat(); }
                        Token::RightParen => {}
                        _ => { return Err(()); }
                    }
                }
```

Reply:
This code is not equivalent, but I like the idea.

Reply:
It isn't? I tried to make it equivalent.

Reply:
Oh right I see it now.